src/test/java/com/amihaiemil/camel/ScalarTest.java

```diff
@@ -82,4 +84,25 @@ public void orphanForbidden() {
             new ArrayList<AbstractNode>(), "orphan"
         );
     }
+
+    /**
+     * Make sure that equals and hash code are reflexive,
+     * symmetric, and transitive.
+     */
+    @Test
+    public void testEqualsAndHashCode() {
+        final String firstValue = "test scalar value #1";
+        final String secondValue = "test scalar value #2";
+        final Scalar<String> firstScalar = new Scalar<String>(
+            Arrays.asList(Mockito.mock(AbstractNode.class)),
+            firstValue
+        );
+        final Scalar<String> secondScalar = new Scalar<String>(
+            Arrays.asList(Mockito.mock(AbstractNode.class)),
+            secondValue
+        );
+        Assert.assertTrue(firstScalar.equals(secondScalar)
```

Comment:
@salikjan This assertion fails, because firstScalar and secondScalar are not equal (look above, how they are declared, the 2 string values do not match)